farmerbot/parser/parser.go

```diff
@@ -0,0 +1,39 @@
+package parser
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/threefoldtech/tfgrid-sdk-go/farmerbot/internal"
+	"gopkg.in/yaml.v3"
+)
+
+// ReadFile reads a file and returns its contents
+func ReadFile(path string) ([]byte, error) {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return content, nil
+}
+
+// ParseIntoConfig parses the configuration
+func ParseIntoConfig(content []byte) (internal.Config, error) {
```

Comment:
I think validation of the configurations needs to get expanded beyond the checking of overlapping nodes ranges
non exhaustive list based on the configurations mentioned in the README.md

- invalid farm ID
- invalid nodes IDs within that farm
- duplicate nodes 
- required parameters e.g in this case would be farmID on the yaml struct.
- overprovision in range 1-4
- invalid datetime format
- invalid threshold values e.g negative numbers
- invalid network